src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -255,23 +271,41 @@ public function save($uuid, array $data)
     public function delete($uuid)
     {
         $document = $this->find($uuid);
+        $webspaceKey = $this->documentInspector->getWebspace($document);
         $this->documentManager->remove($document);
 
+        $this->documentDomainEventCollector->collect(new CustomUrlRemovedEvent(
+            $document->getUuid(),
+            $document->getTitle(),
+            $webspaceKey
+        ));
+
         return $document;
     }
 
     public function deleteRoute($webspaceKey, $uuid)
     {
         $routeDocument = $this->findRoute($uuid);
 
+        $customUrlDocument = $routeDocument->getTargetDocument();
+        while ($customUrlDocument instanceof RouteDocument) {
+            $customUrlDocument = $customUrlDocument->getTargetDocument();
+        }
+
         if (!$routeDocument->isHistory()) {
             $route = PathHelper::relativizePath($routeDocument->getPath(), $this->getRoutesPath($webspaceKey));
 
-            throw new RouteNotRemovableException($route, $routeDocument, $routeDocument->getTargetDocument());
+            throw new RouteNotRemovableException($route, $routeDocument, $customUrlDocument);
```

Comment:
Okay can you explain what is happening here with the while above? Maybe add a comment to the `while` method